server/spec/qpid_spec.rb

```diff
@@ -4,136 +4,198 @@
-
-     assert_mode_reason("NORMAL", "QPID_UP")
-     
-     puts "Candlepin in Normal mode. Creating owner"
-     expect(create_owner random_string('test_owner')).to have_key("created")
-     sleep 15
-     msgs = @cq.receive
-     msgs.length.should == 1
-     msgs[0].subject.should == 'owner.created'
+  it 'should transition to SUSPEND mode' do
+    assert_mode("NORMAL")
+
+    # puts "Candlepin is in Normal mode. Stopping Qpid"
+    @cq.stop
+
+    status = wait_for_mode("SUSPEND")
+    assert_mode_and_reason("SUSPEND", "QPID_DOWN", status)
+
+    # puts "Candlepin is in Suspend mode. Creating owner"
+
+    begin
+      create_owner random_string('test_owner')
+      fail("Expected error did not occur")
+    rescue RestClient::ServiceUnavailable => un
+      displayMessage = JSON.parse(un.response.body)["displayMessage"]
+      displayMessage.should == 'Candlepin is in Suspend mode, please check /status resource to get more details'
+    end
+
+    # puts "Starting Qpid"
+    @cq.start
+
+    status = wait_for_mode("NORMAL")
+    assert_mode_and_reason("NORMAL", "QPID_UP", status)
+
+    # puts "Candlepin in Normal mode. Creating owner"
+    expect(create_owner random_string('test_owner')).to have_key("created")
+
+    msgs = @cq.receive
+    expect(msgs.length).to eq(1)
+    msgs[0].subject.should == 'owner.created'
   end
 
+  it 'should transition to NORMAL mode' do
+    assert_mode("NORMAL")
+
+    # puts "Stopping Qpid and creating owner..."
+    @cq.stop
+    create_owner random_string("test")
+
+    status = wait_for_mode("SUSPEND")
+    assert_mode_and_reason("SUSPEND", "QPID_DOWN", status)
 
-  it 'reconnect should work' do
-     @cq.stop
-     puts "Qpid stopped, creating owner"
-     create_owner random_string("test")
-     sleep 10
-     
-     @cq.start 
-     puts "Qpid restarted, querying queues"
-     sleep 1
-     msgs = @cq.receive
-     msgs.length.should == 0
-    
-     puts "Nothing found in the queues. Waiting for reconnect"
-     
-     sleep 20 
-     create_owner random_string("test")
-     sleep 3
-     msgs = @cq.receive
-     msgs.length.should == 1
-     msgs[0].subject.should == 'owner.created'
+    # puts "Restarting Qpid and querying queues..."
+    @cq.start
+
+    # puts "Waiting for Candlepin to reconnect to Qpid..."
+    status = wait_for_mode("NORMAL")
+    assert_mode("NORMAL", status)
+
+    # puts "Reconnected. Creating owner and waiting for the message..."
+    create_owner random_string("test")
+
+    msgs = @cq.receive
+    msgs.length.should == 1
+    msgs[0].subject.should == 'owner.created'
   end
 
-  it 'Candlepin startup without qpid ends in suspend mode' do
-     assert_mode("NORMAL")
-     stop_tomcat
-     @cq.stop
-     start_tomcat
-     sleep 10
-     assert_mode("SUSPEND")
-     @cq.start
-     sleep 12
-     assert_mode("NORMAL")
+  it 'should start in SUSPEND mode without Qpid' do
+    def start_tomcat()
+       `sudo systemctl start tomcat || sudo supervisorctl start tomcat`
```

Comment:
This should check existence of supervisorctl & use that instead of systemctl. Otherwise, systemctl usage will raise dbus erros in the logs when run inside a container.

Reply:
bashism for this:
```
type systemctl &> /dev/null && sudo systemctl start tomcat || sudo supervisorctl stop tomcat
```
there is probably a nicer lookin' ruby version of this

Reply:
Done.